external_search.py

```diff
@@ -17,12 +17,25 @@ class ExternalSearchIndex(object):
     work_document_type = 'work-type'
     __client = None
 
+    CURRENT_ALIAS_SUFFIX = '-current'
+    VERSION_RE = re.compile('-v([0-9]+)$')
+
+    @classmethod
+    def reset(cls):
+        """Resets the __client object to None so a new configuration
+        can be applied during object initialization.
+
+        This method is only intended for use in testing.
+        """
+        cls.__client = None
+
     def __init__(self, url=None, works_index=None):
     
         self.log = logging.getLogger("External search index")
 
         # By default, assume that there is no search index.
         self.works_index = None
+        from_configuration = False
```

Comment:
I'm not sure whether or not a config file was used is the right way to determine if we should create the alias.

The constructor could always create the alias if it doesn't exist yet, but I think we could separate out the code that moves the alias to a new index version, so it never happens as part of the constructor. I'd like to have that happen in a separate script anyway, not when you're running the app or creating a new index.